rust/arrow/src/compute/kernels/sort.rs

```diff
@@ -149,9 +167,13 @@ where
         .collect::<Vec<(u32, T::Native)>>();
     let mut nulls = null_indices;
     if !options.descending {
-        valids.sort_by_key(|a| a.1);
+        valids.sort_by(|a, b| a.1.partial_cmp(&b.1).unwrap_or_else(|| Ordering::Greater));
```

Comment:
i am kind of curious, have you done any benchmark to see if changing to partial_cmp with unwrap_or_else results in any performance difference?

Reply:
I haven't, I can try it out when I get a chance, as I'm also curious of what the difference would be